services/configuration/src/cmd.rs

```diff
@@ -0,0 +1,215 @@
+
+        let mut values: BTreeMap<String, Value> = context.get(keys::SERVICES_CONFIG).expect(
+            "Expected services_config \
+             in context.",
+        );
+
+        let byzantine_majority_count =
+            State::byzantine_majority_count(validators_count as usize) as u16;
+
+        validate_majority_count(majority_count, validators_count, byzantine_majority_count)
+            .unwrap();
+
+        if let Some(majority_count) = majority_count {
+            values.extend(
+                vec![(
+                    "majority_count".to_owned(),
+                    Value::try_from(majority_count).unwrap(),
+                )].into_iter(),
+            );
+        };
+
+        context.set(keys::SERVICES_CONFIG, values);
+        Ok(context)
+    }
+}
+
+pub struct Finalize;
+
+impl CommandExtension for Finalize {
+    fn args(&self) -> Vec<Argument> {
+        vec![]
+    }
+
+    fn execute(&self, mut context: Context) -> Result<Context, failure::Error> {
+        let mut node_config: NodeConfig = context.get(keys::NODE_CONFIG).unwrap();
+        let common_config = context.get(keys::COMMON_CONFIG).unwrap();
+
+        // Local config section
+        let majority_count =
+            if let Some(majority_count) = common_config.services_config.get("majority_count") {
+                Value::try_into(majority_count.clone()).unwrap_or_default()
+            } else {
+                Default::default()
+            };
+
+        node_config.services_configs.insert(
+            "configuration_service".to_owned(),
+            Value::try_from(ConfigurationServiceConfig { majority_count })
+                .expect("Could not serialize configuration service config"),
+        );
+        context.set(keys::NODE_CONFIG, node_config);
+        Ok(context)
+    }
+}
+
+pub struct GenerateTestnet;
+
+impl CommandExtension for GenerateTestnet {
+    fn args(&self) -> Vec<Argument> {
+        vec![Argument::new_named(
+            "MAJORITY_COUNT",
+            false,
+            "Number of votes required to commit new configuration",
+            None,
+            "majority-count",
+            false,
+        )]
+    }
+
+    fn execute(&self, mut context: Context) -> Result<Context, failure::Error> {
+        let validators_count: u16 = context.arg("COUNT").expect("count as int");
+        let start_port = context
+            .arg::<u16>("START_PORT")
+            .unwrap_or(DEFAULT_EXONUM_LISTEN_PORT);
+
+        if validators_count == 0 {
+            panic!("Can't generate testnet with zero nodes count.");
+        }
+
+        let majority_count = context.arg::<u16>("MAJORITY_COUNT").ok();
+
+        let byzantine_majority_count =
+            State::byzantine_majority_count(validators_count as usize) as u16;
+        validate_majority_count(majority_count, validators_count, byzantine_majority_count)
+            .unwrap();
+
+        let configs = generate_testnet_config(validators_count, start_port, majority_count);
```

Comment:
This testnet extension probably wouldn't work with other services.

Reply:
Does majority count will recalculate when new validator will be added? - No